factcast-store/src/main/java/org/factcast/store/internal/query/CurrentStatementHolder.java

```diff
@@ -33,7 +33,13 @@ public void close() {
       if (statement != null) {
         log.info("Canceling statement " + statement);
         try {
+
           statement.cancel();
+
+          // we have to roll back the tx on the underlying connection
+          // if we do not end the transaction, statements are cancelled but still "idle in
+          // transaction" and so block further actions like wiping between tests
+          statement.getConnection().rollback();
```

Comment:
wasCanceled should be set to true **before** the rollback is triggered, so that in case of an exception the boolean is set, right?

Reply:
I was thinking of moving it to a finally block 👋

Reply:
:laughing:

Reply:
even better

Reply:
done :)